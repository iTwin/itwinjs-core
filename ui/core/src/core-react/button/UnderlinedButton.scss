/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "../style/index.scss";
@import "../inputs/index.scss";

$color: $buic-foreground-primary;
$hoverColor: $buic-foreground-primary-tone;

.core-underlined-button {
  background: transparent;
  color: $color;
  border: 0px solid $color;
  border-bottom-width: 1px;

  width: fit-content;

  cursor: pointer;

  &:hover {
    color: $hoverColor;
    border-color: $hoverColor;
  }

  &:focus {
    color: $hoverColor;
    border-color: $hoverColor;
    background-color: $buic-background-focus-overlay;
    outline: none;
  }

}