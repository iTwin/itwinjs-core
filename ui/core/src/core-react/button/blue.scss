/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "../style/index";

@import "./button";
@import "./disabled";


@mixin uicore-buttons-blue {

  @include uicore-buttons-button;

  background:         $buic-accessory-primary;
  color:              $buic-accessory-hollow;

  &:hover {
    background-color:   $uicore-blue-5;
  }

  &:active {
    background-color:   $uicore-blue-4;
  }
}
