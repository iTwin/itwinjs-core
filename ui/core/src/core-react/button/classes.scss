/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "./index";

.uicore-buttons-blue { @include uicore-buttons-blue; }
.uicore-buttons-blue-large { @include uicore-buttons-blue-large; }
.uicore-buttons-disabled { @include uicore-buttons-disabled; }
.uicore-buttons-disabled-large { @include uicore-buttons-disabled-large; }
.uicore-buttons-hollow { @include uicore-buttons-hollow; }
.uicore-buttons-hollow-large { @include uicore-buttons-hollow-large; }
.uicore-buttons-primary { @include uicore-buttons-primary; }
.uicore-buttons-primary-large { @include uicore-buttons-primary-large; }
