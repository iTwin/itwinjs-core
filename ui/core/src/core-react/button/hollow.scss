/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "../style/index";
@import "../inputs/variables";

@import "./button";
@import "./disabled";

@mixin uicore-buttons-hollow {

    @include uicore-buttons-button;

    padding:            $uicore-button-padding-bordered $uicore-bordered-padding;
    border-color:       $buic-foreground-disabled;
    background-color:   transparent;
    color:              $buic-text-color;

    &:hover {

      border-color:       $buic-foreground-body;
      color:              $buic-foreground-activehover;
    }

    &:active, &:focus {

      border-color:       $buic-foreground-activehover;
    }
}
