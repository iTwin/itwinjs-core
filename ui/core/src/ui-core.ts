/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/

// cSpell:ignore focustrap imagecheckbox iconinput hocs numberinput numericinput

export * from "./ui-core/UiCore";

export * from "./ui-core/badge/Badge";
export * from "./ui-core/badge/BadgeUtilities";
export * from "./ui-core/badge/BetaBadge";
export * from "./ui-core/badge/NewBadge";

export * from "./ui-core/base/Div";
export * from "./ui-core/base/DivWithOutsideClick";
export * from "./ui-core/base/Centered";
export * from "./ui-core/base/FillCentered";
export * from "./ui-core/base/FlexWrapContainer";
export * from "./ui-core/base/PointerEvents";
export * from "./ui-core/base/ScrollView";

export * from "./ui-core/button/Button";
export * from "./ui-core/button/UnderlinedButton";

export * from "./ui-core/checklistbox/CheckListBox";

export * from "./ui-core/contextmenu/ContextMenu";
export * from "./ui-core/contextmenu/ContextMenuDirection";
export * from "./ui-core/contextmenu/ContextMenuDivider";
export * from "./ui-core/contextmenu/ContextMenuItem";
export * from "./ui-core/contextmenu/ContextSubMenu";
export * from "./ui-core/contextmenu/GlobalContextMenu";
export * from "./ui-core/contextmenu/PopupContextMenu";

export * from "./ui-core/cube/Cube";

export * from "./ui-core/dialog/Dialog";

export * from "./ui-core/elementseparator/ElementSeparator";

export * from "./ui-core/enums/Alignment";
export * from "./ui-core/enums/CheckBoxState";
export * from "./ui-core/enums/Orientation";
export * from "./ui-core/enums/SortDirection";
export * from "./ui-core/enums/TimeFormat";

export * from "./ui-core/expandable/ExpandableList";
export * from "./ui-core/expandable/ExpandableBlock";

export * from "./ui-core/focus/ItemKeyboardNavigator";
export * from "./ui-core/focustrap/FocusTrap";

export * from "./ui-core/form/Field";
export * from "./ui-core/form/Form";

export * from "./ui-core/hocs/withIsPressed";
export * from "./ui-core/hocs/withOnOutsideClick";
export * from "./ui-core/hocs/withTimeout";

export * from "./ui-core/icons/IconComponent";
export * from "./ui-core/icons/SvgPath";
export * from "./ui-core/icons/SvgSprite";
export * from "./ui-core/icons/WebFontIcon";

export * from "./ui-core/autosuggest/AutoSuggest";

export * from "./ui-core/checkbox/Checkbox";

export * from "./ui-core/imagecheckbox/ImageCheckBox";

export * from "./ui-core/inputs/Input";
export * from "./ui-core/inputs/InputLabel";
export * from "./ui-core/inputs/InputStatus";
export * from "./ui-core/inputs/iconinput/IconInput";
export * from "./ui-core/inputs/LabeledComponentProps";
export * from "./ui-core/inputs/LabeledInput";
export * from "./ui-core/inputs/LabeledTextarea";
export * from "./ui-core/inputs/numberinput/NumberInput";
export * from "./ui-core/inputs/numericinput/NumericInput";
export * from "./ui-core/inputs/numericinput/ReactNumericInput";
export * from "./ui-core/inputs/Textarea";

export * from "./ui-core/listbox/Listbox";

export * from "./ui-core/loading/LoadingBar";
export * from "./ui-core/loading/LoadingPrompt";
export * from "./ui-core/loading/LoadingSpinner";
export * from "./ui-core/loading/LoadingStatus";
export * from "./ui-core/loading/Spinner";

export * from "./ui-core/messagebox/MessageBox";

export * from "./ui-core/notification/MessageRenderer";
export * from "./ui-core/notification/MessageType";

export * from "./ui-core/popup/Popup";

export * from "./ui-core/progress-indicators/ProgressBar";
export * from "./ui-core/progress-indicators/ProgressSpinner";

export * from "./ui-core/radialmenu/RadialMenu";
export * from "./ui-core/radialmenu/Annulus";
export * from "./ui-core/radio/Radio";

export * from "./ui-core/select/LabeledSelect";
export * from "./ui-core/select/Select";
export * from "./ui-core/select/ThemedSelect";
export * from "./ui-core/select/LabeledThemedSelect";
export * from "./ui-core/searchbox/SearchBox";

export * from "./ui-core/settings/SettingsManager";
export * from "./ui-core/settings/SettingsContainer";

export * from "./ui-core/slider/Slider";

export * from "./ui-core/splitbutton/SplitButton";

export * from "./ui-core/tabs/HorizontalTabs";
export * from "./ui-core/tabs/VerticalTabs";
export * from "./ui-core/tabs/Tabs";

export * from "./ui-core/text/BodyText";
export * from "./ui-core/text/BlockText";
export * from "./ui-core/text/DisabledText";
export * from "./ui-core/text/FilteredText";
export * from "./ui-core/text/Headline";
export * from "./ui-core/text/LeadingText";
export * from "./ui-core/text/LeadingText2";
export * from "./ui-core/text/MutedText";
export * from "./ui-core/text/SmallText";
export * from "./ui-core/text/Subheading";
export * from "./ui-core/text/Subheading2";
export * from "./ui-core/text/StyledText";
export * from "./ui-core/text/TextProps";
export * from "./ui-core/text/Title";
export * from "./ui-core/text/Title2";

export * from "./ui-core/tiles/FeaturedTile";
export * from "./ui-core/tiles/MinimalFeaturedTile";
export * from "./ui-core/tiles/MinimalTile";
export * from "./ui-core/tiles/Tile";

export * from "./ui-core/toggle/Toggle";
export * from "./ui-core/toggle/LabeledToggle";

export * from "./ui-core/tooltip/Tooltip";

export { ExpansionToggle, ExpansionToggleProps } from "./ui-core/tree/ExpansionToggle";
export { TreeBranch, TreeBranchProps } from "./ui-core/tree/Branch";
export { TreeNode, TreeNodeProps, NodeCheckboxProps, NodeCheckboxRenderer, NodeCheckboxRenderProps } from "./ui-core/tree/Node";
export { Tree, TreeProps } from "./ui-core/tree/Tree";
export { TreeNodePlaceholder, TreeNodePlaceholderProps } from "./ui-core/tree/Placeholder";

export * from "./ui-core/uisettings/UiSetting";
export * from "./ui-core/uisettings/UiSettings";
export * from "./ui-core/uisettings/LocalUiSettings";
export * from "./ui-core/uisettings/SessionUiSettings";

export * from "./ui-core/utils/IconHelper";
export * from "./ui-core/utils/Point";
export * from "./ui-core/utils/Props";
export * from "./ui-core/utils/Rectangle";
export * from "./ui-core/utils/Size";
export * from "./ui-core/utils/Timer";
export * from "./ui-core/utils/UiEvent";
export * from "./ui-core/utils/flattenChildren";
export * from "./ui-core/utils/getBestBWContrastColor";
export * from "./ui-core/utils/getCssVariable";
export * from "./ui-core/utils/getDisplayName";
export * from "./ui-core/utils/getUserColor";
export * from "./ui-core/utils/shallowDiffers";
export * from "./ui-core/utils/typeUtils";
export * from "./ui-core/utils/isPromiseLike";
export * from "./ui-core/utils/ScrollPositionMaintainer";
export * from "./ui-core/utils/hooks/useDisposable";
export * from "./ui-core/utils/hooks/useEffectSkipFirst";
export * from "./ui-core/utils/hooks/ResizeObserverPolyfill";
export * from "./ui-core/utils/hooks/useOnOutsideClick";
export * from "./ui-core/utils/hooks/useProximityToMouse";
export * from "./ui-core/utils/hooks/useRefEffect";
export * from "./ui-core/utils/hooks/useRefs";
export * from "./ui-core/utils/hooks/useRefState";
export * from "./ui-core/utils/hooks/useResizeObserver";
export * from "./ui-core/utils/hooks/useTargeted";
export * from "./ui-core/utils/hooks/useWidgetOpacityContext";

/** @docs-package-description
 * The ui-core package contains general purpose React components, such as Dialog, MessageBox, SearchBox, RadialMenu and SplitButton.
 * For more information, see [learning about ui-core]($docs/learning/ui/core/index.md).
 */
/**
 * @docs-group-description AutoSuggest
 * Component for input with an auto-suggestion dropdown.
 */
/**
 * @docs-group-description Base
 * Low-level classes and components for building application UI.
 */
/**
 * @docs-group-description Button
 * Components for working with various Buttons.
 */
/**
 * @docs-group-description Checkbox
 * Component is a wrapper for the `<input type="checkbox">` HTML element.
 */
/**
 * @docs-group-description CheckListBox
 * Components for working with a Check listbox.
 */
/**
 * @docs-group-description Common
 * Common classes and enums used across various UI components.
 */
/**
 * @docs-group-description ContextMenu
 * Components for working with a Context Menu.
 */
/**
 * @docs-group-description Cube
 * Component for 3D Cube.
 */
/**
 * @docs-group-description Dialog
 * Components for working with a Dialog or MessageBox.
 */
/**
 * @docs-group-description ElementSeparator
 * Components for working with a ElementSeparator.
 */
/**
 * @docs-group-description Expandable
 * Components for working with a ExpandableBlock or ExpandableList.
 */
/**
 * @docs-group-description Form
 * Components used to create a Form using supplied properties to specify fields.
 */
/**
 * @docs-group-description Icon
 * Component that renders ui-core icon when given an icon name or SVG source.
 */
/**
 * @docs-group-description Inputs
 * Components for working with input controls, such as Input, IconInput, NumberInput and Textarea.
 */
/**
 * @docs-group-description Loading
 * Components for working with Loading spinners and progress indicators and bars.
 */
/**
 * @docs-group-description Notification
 * Components for working with messages and tooltips.
 */
/**
 * @docs-group-description Popup
 * Components for working with a Popup.
 */
/**
 * @docs-group-description RadialMenu
 * Components for working with a RadialMenu.
 */
/**
 * @docs-group-description Radio
 * Component is a wrapper for the `<input type="radio">` HTML element.
 */
/**
 * @docs-group-description SearchBox
 * Components for working with a SearchBox.
 */
/**
 * @docs-group-description Select
 * Component is a wrapper for the `<select>` HTML element.
 */
/**
 * @docs-group-description Settings
 * Manager and UI Components that allow users to modify settings for different packages and extensions.
 */
/**
 * @docs-group-description Slider
 * Component displays a range slider with thumbs for changing the value.
 */
/**
 * @docs-group-description SplitButton
 * Components for working with a SplitButton.
 */
/**
 * @docs-group-description Tabs
 * Components for working with horizontal or vertical tabs.
 */
/**
 * @docs-group-description Text
 * Components for working with styled text.
 */
/**
 * @docs-group-description Tiles
 * Components for a container rendering elements that can be grouped together.
 */
/**
 * @docs-group-description Toggle
 * Components for working with a Toggle switch.
 */
/**
 * @docs-group-description Tooltip
 * Components for working with a Tooltip.
 */
/**
 * @docs-group-description Tree
 * Presentation React components for working with a Tree.
 */
/**
 * @docs-group-description UiSettings
 * Interfaces and classes for working with persistent UI settings.
 */
/**
 * @docs-group-description Utilities
 * Various utility classes, functions and React hooks for working with a UI.
 */
