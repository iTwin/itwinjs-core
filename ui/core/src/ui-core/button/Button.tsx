/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module Button
 */

import classnames from "classnames";
import * as React from "react";
import { CommonProps } from "../utils/Props";

/** Sizes for [[Button]] component
 * @public
 */
export enum ButtonSize {
  Default = "",
  Large = "large",
}

/** Types for [[Button]] component
 * @public
 */
export enum ButtonType {
  Primary = "primary",
  Blue = "blue",
  Disabled = "disabled",
  Hollow = "hollow",
}

/** Properties for [[Button]] component
 * @public
 */
export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement>, CommonProps {
  /** Allow ID to be passed to Button */
  id?: string;
  /** Default and large sizes */
  size?: ButtonSize;
  /** 4 styles to tweak the content of the button */
  buttonType?: ButtonType;
  /** A function to be run when the element is clicked */
  onClick?: ((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void);
}

/** Generic button component
 * @public
 */
export class Button extends React.PureComponent<ButtonProps> {
  public override render() {
    const { buttonType, size, className, style, onClick, ...props } = this.props;

    let typeClassName = getButtonTypeClassName(buttonType);

    if (size === ButtonSize.Large)
      typeClassName += "-large";

    return <button {...props} className={classnames(typeClassName, className)} style={style} onClick={onClick} />;
  }
}

/** @internal */
export function getButtonTypeClassName(buttonType?: ButtonType): string {
  let typeClassName: string;
  switch (buttonType) {
    case ButtonType.Blue:
      typeClassName = "uicore-buttons-blue";
      break;
    case ButtonType.Disabled:
      typeClassName = "uicore-buttons-disabled";
      break;
    case ButtonType.Hollow:
      typeClassName = "uicore-buttons-hollow";
      break;
    case ButtonType.Primary:
    default:
      typeClassName = "uicore-buttons-primary";
      break;
  }
  return typeClassName;
}
