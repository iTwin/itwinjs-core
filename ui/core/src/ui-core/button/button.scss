/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "./variables";
@import "~@itwin/itwinui-css/scss/style/speed";
@import "~@itwin/itwinui-css/scss/style/typography";
@import "~@itwin/itwinui-css/scss/style/variables";

// cspell:ignore unbordered

/// Mixins ---------------------------------------------------------------------

@mixin uicore-buttons-button {

  display:                inline-block;

  padding:                $uicore-button-padding $uicore-unbordered-padding;
  margin:                 $uicore-inputs-margin 0;
  border-radius:          $iui-border-radius;
  line-height:            $uicore-button-line-height;
  outline:                none;
  font-family:            $iui-font-family;
  font-size:              $iui-font-size;
  text-decoration:        none;
  cursor:                 pointer;
  box-sizing:             border-box;
  box-shadow:             transparent 0px 1px 1px inset;
  border:                 1px solid transparent;
  transition:             all $iui-speed-fast ease-in-out;

  &:active,
  &:focus {

    @include uicore-buttons-button-focused;
  }

  &[disabled],
  &[disabled]:hover,
  &[disabled]:active,
  &[disabled]:focus {

    @include uicore-buttons-button-disabled;
  }
}

@mixin uicore-buttons-button-focused {

  box-shadow:           $uicore-button-focus-boxshadow;
}

@mixin uicore-buttons-button-unbordered {

  border-color:         transparent;
  box-shadow:           transparent 0px 1px 1px inset;
}

@mixin uicore-buttons-button-disabled {

  opacity:                $uicore-opacity-disabled;
  background-color:       $uicore-gray;
  border:                 0;
  color:                  $uicore-white;
  cursor:                 not-allowed;
}

@mixin uicore-buttons-large-size {

  padding:                $uicore-inputs-padding-large $uicore-unbordered-padding-large;
  margin:                 $uicore-inputs-margin-large 0;
  font-size:              $iui-font-size-leading;
  font-weight:            $iui-font-weight-normal;
}

@mixin uicore-buttons-large-bordered-size {

  padding:                $uicore-button-padding-bordered-large $uicore-bordered-padding-large;
  margin:                 $uicore-inputs-margin-large 0;
  font-size:              $iui-font-size-leading;
  font-weight:            $iui-font-weight-normal;
}
