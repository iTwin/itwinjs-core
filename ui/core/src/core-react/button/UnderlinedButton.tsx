/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module Button
 */

import "./UnderlinedButton.scss";
import classnames from "classnames";
import * as React from "react";
import { SpecialKey } from "@itwin/appui-abstract";

/** Properties for the [[UnderlinedButton]] React component
 * @public
 */
export interface UnderlinedButtonProps {
  /** String that will be rendered by the button */
  children: string | React.ReactNode;
  /** Additional className */
  className?: string;
  /** Title of the button */
  title?: string;
  /** Callback to onClick event */
  onClick?: (e: React.MouseEvent) => void;
  /** Callback to activate */
  onActivate?: () => void;
}

/** A React component that makes text clickable and underlined
 * @public
 */
export function UnderlinedButton(props: UnderlinedButtonProps) {
  const handleKeyUp = React.useCallback((event: React.KeyboardEvent) => {
    const key = event.key;

    switch (key) {
      case SpecialKey.Enter:
      case SpecialKey.Space:
        props.onActivate && props.onActivate();
        break;
    }
  }, [props]);

  const handleClick = React.useCallback((e: React.MouseEvent) => {
    props.onClick && props.onClick(e);
    props.onActivate && props.onActivate();
  }, [props]);

  const className = classnames(
    "core-underlined-button",
    props.className ? props.className : undefined,
  );

  return (
    <span
      className={className}
      title={props.title}
      onClick={handleClick}
      onKeyUp={handleKeyUp}
      tabIndex={0}
      role="link"
    >
      {props.children}
    </span>
  );
}
