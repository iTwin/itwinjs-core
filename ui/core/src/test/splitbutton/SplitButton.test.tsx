/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
import { mount, shallow } from "enzyme";
import { expect } from "chai";
import * as sinon from "sinon";
import * as React from "react";
import { SplitButton } from "../../ui-core";
import { RelativePosition, SpecialKey } from "@bentley/ui-abstract";
import { ButtonType } from "../../ui-core/button/Button";

/* eslint-disable deprecation/deprecation */

describe("<SplitButton />", () => {
  it("should render", () => {
    const wrapper = mount(<SplitButton label="test" />);
    wrapper.unmount();
  });

  it("renders correctly", () => {
    shallow(<SplitButton label="test" />).should.matchSnapshot();
  });

  it("renders with icon correctly", () => {
    shallow(<SplitButton label="test" icon="icon-placeholder" />).should.matchSnapshot();
  });

  it("renders with drawBorder correctly", () => {
    shallow(<SplitButton label="test" drawBorder />).should.matchSnapshot();
  });

  it("renders with popupPosition correctly", () => {
    shallow(<SplitButton label="test" popupPosition={RelativePosition.BottomLeft} />).should.matchSnapshot();
  });

  it("renders with buttonType=Blue correctly", () => {
    shallow(<SplitButton label="test" buttonType={ButtonType.Blue} />).should.matchSnapshot();        // eslint-disable-line deprecation/deprecation
  });
  it("renders with buttonType=Disabled correctly", () => {
    shallow(<SplitButton label="test" buttonType={ButtonType.Disabled} />).should.matchSnapshot();    // eslint-disable-line deprecation/deprecation
  });
  it("renders with buttonType=Hollow correctly", () => {
    shallow(<SplitButton label="test" buttonType={ButtonType.Hollow} />).should.matchSnapshot();      // eslint-disable-line deprecation/deprecation
  });
  it("renders with buttonType=Primary correctly", () => {
    shallow(<SplitButton label="test" buttonType={ButtonType.Primary} />).should.matchSnapshot();     // eslint-disable-line deprecation/deprecation
  });

  it("handles keydown/up correctly", () => {
    const wrapper = mount<SplitButton>(<SplitButton label="test" />);
    wrapper.find(".core-split-button").at(0).simulate("keyup", { key: SpecialKey.ArrowDown });
    expect(wrapper.state().expanded).to.be.true;

    wrapper.find(".core-split-button").at(0).simulate("keyup", { keyCode: 0 });
    wrapper.unmount();
  });

  it("calls onExecute on Enter keyup", () => {
    const spyMethod = sinon.spy();
    const wrapper = mount(<SplitButton label="test" onExecute={spyMethod} />);
    wrapper.find(".core-split-button").at(0).simulate("keyup", { key: SpecialKey.Enter });
    spyMethod.calledOnce.should.true;
    wrapper.unmount();
  });

  it("handles click on arrow correctly", () => {
    const wrapper = mount<SplitButton>(<SplitButton label="test" />);
    wrapper.find(".core-split-button-arrow").at(0).simulate("click");
    expect(wrapper.state().expanded).to.be.true;
    wrapper.find(".core-split-button-arrow").at(0).simulate("click");
    expect(wrapper.state().expanded).to.be.false;
    wrapper.unmount();
  });

  it("handles menu close correctly", () => {
    const wrapper = mount<SplitButton>(<SplitButton label="test" />);
    wrapper.find(".core-split-button-arrow").at(0).simulate("click");
    expect(wrapper.state().expanded).to.be.true;
    wrapper.find(".core-context-menu").at(0).simulate("click", { target: document.getElementsByClassName(".core-split-button-arrow")[0] });
    expect(wrapper.state().expanded).to.be.false;
    wrapper.unmount();
  });

  it("handles initialExpanded prop correctly", () => {
    const wrapper = mount<SplitButton>(<SplitButton label="test" initialExpanded={true} />);
    expect(wrapper.state().expanded).to.be.true;
    wrapper.unmount();
  });

});
