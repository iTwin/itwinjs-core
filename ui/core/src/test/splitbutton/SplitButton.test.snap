// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplitButton /> renders correctly 1`] = `
<div
  className="core-split-button core-split-button-default-colors"
  data-testid="core-split-button-root"
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="core-split-button-label"
    data-testid="core-split-button-label"
    role="button"
    tabIndex={-1}
  >
     
    test
  </div>
  <div
    className="core-split-button-divider"
  />
  <div
    className="core-split-button-arrow"
    onClick={[Function]}
    role="button"
    tabIndex={-1}
  >
    <div
      className="core-split-button-arrow-icon icon icon-chevron-down"
    />
    <PopupContextMenu
      animate={false}
      autoflip={false}
      isOpen={false}
      onEsc={[Function]}
      onOutsideClick={[Function]}
      onSelect={[Function]}
      position={6}
      selectedIndex={0}
      target={null}
    />
  </div>
</div>
`;

exports[`<SplitButton /> renders with buttonType=Blue correctly 1`] = `
<div
  className="core-split-button uicore-buttons-blue"
  data-testid="core-split-button-root"
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="core-split-button-label"
    data-testid="core-split-button-label"
    role="button"
    tabIndex={-1}
  >
     
    test
  </div>
  <div
    className="core-split-button-divider"
  />
  <div
    className="core-split-button-arrow"
    onClick={[Function]}
    role="button"
    tabIndex={-1}
  >
    <div
      className="core-split-button-arrow-icon icon icon-chevron-down"
    />
    <PopupContextMenu
      animate={false}
      autoflip={false}
      isOpen={false}
      onEsc={[Function]}
      onOutsideClick={[Function]}
      onSelect={[Function]}
      position={6}
      selectedIndex={0}
      target={null}
    />
  </div>
</div>
`;

exports[`<SplitButton /> renders with buttonType=Disabled correctly 1`] = `
<div
  className="core-split-button uicore-buttons-disabled"
  data-testid="core-split-button-root"
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="core-split-button-label"
    data-testid="core-split-button-label"
    role="button"
    tabIndex={-1}
  >
     
    test
  </div>
  <div
    className="core-split-button-divider"
  />
  <div
    className="core-split-button-arrow"
    onClick={[Function]}
    role="button"
    tabIndex={-1}
  >
    <div
      className="core-split-button-arrow-icon icon icon-chevron-down"
    />
    <PopupContextMenu
      animate={false}
      autoflip={false}
      isOpen={false}
      onEsc={[Function]}
      onOutsideClick={[Function]}
      onSelect={[Function]}
      position={6}
      selectedIndex={0}
      target={null}
    />
  </div>
</div>
`;

exports[`<SplitButton /> renders with buttonType=Hollow correctly 1`] = `
<div
  className="core-split-button uicore-buttons-hollow"
  data-testid="core-split-button-root"
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="core-split-button-label"
    data-testid="core-split-button-label"
    role="button"
    tabIndex={-1}
  >
     
    test
  </div>
  <div
    className="core-split-button-divider"
  />
  <div
    className="core-split-button-arrow"
    onClick={[Function]}
    role="button"
    tabIndex={-1}
  >
    <div
      className="core-split-button-arrow-icon icon icon-chevron-down"
    />
    <PopupContextMenu
      animate={false}
      autoflip={false}
      isOpen={false}
      onEsc={[Function]}
      onOutsideClick={[Function]}
      onSelect={[Function]}
      position={6}
      selectedIndex={0}
      target={null}
    />
  </div>
</div>
`;

exports[`<SplitButton /> renders with buttonType=Primary correctly 1`] = `
<div
  className="core-split-button uicore-buttons-primary"
  data-testid="core-split-button-root"
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="core-split-button-label"
    data-testid="core-split-button-label"
    role="button"
    tabIndex={-1}
  >
     
    test
  </div>
  <div
    className="core-split-button-divider"
  />
  <div
    className="core-split-button-arrow"
    onClick={[Function]}
    role="button"
    tabIndex={-1}
  >
    <div
      className="core-split-button-arrow-icon icon icon-chevron-down"
    />
    <PopupContextMenu
      animate={false}
      autoflip={false}
      isOpen={false}
      onEsc={[Function]}
      onOutsideClick={[Function]}
      onSelect={[Function]}
      position={6}
      selectedIndex={0}
      target={null}
    />
  </div>
</div>
`;

exports[`<SplitButton /> renders with drawBorder correctly 1`] = `
<div
  className="core-split-button core-split-button-default-colors core-split-button-border"
  data-testid="core-split-button-root"
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="core-split-button-label core-split-button-border"
    data-testid="core-split-button-label"
    role="button"
    tabIndex={-1}
  >
     
    test
  </div>
  <div
    className="core-split-button-divider core-split-button-border"
  />
  <div
    className="core-split-button-arrow"
    onClick={[Function]}
    role="button"
    tabIndex={-1}
  >
    <div
      className="core-split-button-arrow-icon icon icon-chevron-down"
    />
    <PopupContextMenu
      animate={false}
      autoflip={false}
      isOpen={false}
      onEsc={[Function]}
      onOutsideClick={[Function]}
      onSelect={[Function]}
      position={6}
      selectedIndex={0}
      target={null}
    />
  </div>
</div>
`;

exports[`<SplitButton /> renders with icon correctly 1`] = `
<div
  className="core-split-button core-split-button-default-colors"
  data-testid="core-split-button-root"
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="core-split-button-label"
    data-testid="core-split-button-label"
    role="button"
    tabIndex={-1}
  >
    <Icon
      iconSpec="icon-placeholder"
    />
     
    test
  </div>
  <div
    className="core-split-button-divider"
  />
  <div
    className="core-split-button-arrow"
    onClick={[Function]}
    role="button"
    tabIndex={-1}
  >
    <div
      className="core-split-button-arrow-icon icon icon-chevron-down"
    />
    <PopupContextMenu
      animate={false}
      autoflip={false}
      isOpen={false}
      onEsc={[Function]}
      onOutsideClick={[Function]}
      onSelect={[Function]}
      position={6}
      selectedIndex={0}
      target={null}
    />
  </div>
</div>
`;

exports[`<SplitButton /> renders with popupPosition correctly 1`] = `
<div
  className="core-split-button core-split-button-default-colors"
  data-testid="core-split-button-root"
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="core-split-button-label"
    data-testid="core-split-button-label"
    role="button"
    tabIndex={-1}
  >
     
    test
  </div>
  <div
    className="core-split-button-divider"
  />
  <div
    className="core-split-button-arrow"
    onClick={[Function]}
    role="button"
    tabIndex={-1}
  >
    <div
      className="core-split-button-arrow-icon icon icon-chevron-down"
    />
    <PopupContextMenu
      animate={false}
      autoflip={false}
      isOpen={false}
      onEsc={[Function]}
      onOutsideClick={[Function]}
      onSelect={[Function]}
      position={6}
      selectedIndex={0}
      target={null}
    />
  </div>
</div>
`;
